### PR TITLE
Disable `buildvcs`

### LINKED ISSUE #20055: Docker build fails if gitea is a git submodule
### Description

To reproduce

```console
mkdir test
cd test
git init .
git submodule add https://github.com/go-gitea/gitea
cd gitea
docker build . -t tag
```
output :point_down:
```
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
error obtaining VCS status: exit status 128
	Use -buildvcs=false to disable VCS stamping.
Running go generate...
go: downloading github.com/mattn/go-sqlite3 v1.14.12
make: *** No rule to make target 'modules/public/bindata.go', needed by 'gitea'.  Stop.
The command '/bin/sh -c if [ -n "${GITEA_VERSION}" ]; then git checkout "${GITEA_VERSION}"; fi  && make clean-all build' returned a non-zero code: 2
```

### Gitea Version

cb50375

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

2.36.1

### Operating System

Linux

### How are you running Gitea?

Docker build.

### Database

_No response_

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
- Fixes https://github.com/go-gitea/gitea/issues/20055

### CODE REVIEW

**Review by techknowlogick — DISMISSED** [review]
You can achieve this with goflags env var, or extra goflags. This also disables it for all builds, not just your specific use case.

**Comment by wxiaoguang** [comment]
I think Golang's `buildvcs=true/auto` is not ideal. 

> By default, version control information is stamped into a binary if the main package and the main module containing it are in the repository containing the current directory (if there is a repository)

It doesn't help Gitea's build, and may break reproducible build (even the code the exactly the same, the output differs due to different repository information).

**Comment by techknowlogick** [comment]
Yeah, what @42wim said has changed my pov. I'll drop my review, although I'll suggest adding the flag into the existing goflag env var instead of hardcoding it directly.

File: Makefile
Comment (zeripath) [inline Makefile]: Why aren't you just setting EXTRA_GOFLAGS to use this flag for your specific need. Most people building Gitea do not need this flag and will not need it.

Comment (AbdulrhmnGhanem) [reply Makefile]: Even if the reasons above don't justify the use of this flag, what value does this flag remove from other users? It just resets the build to behave as pre-1.18.

**Comment by AbdulrhmnGhanem** [comment]
> I'll suggest adding the flag into the existing goflag env var instead of hardcoding it directly.

Adding
```makefile
GOFLAGS += -buildvcs=false
```
the build fails with the same error, is this wrong?

**Comment by techknowlogick** [comment]
what about just adding it as `EXTRA_GOFLAGS ?= -buildvcs=false` to https://github.com/go-gitea/gitea/blob/389496bc93791b974246dc10aeccd2332873d6ce/Makefile#L68 ?

It'll be overwritten if someone tries to add their own extra goflags, but I think that something that can be dealt with in a later PR.

**Comment by AbdulrhmnGhanem** [comment]
> what about just adding it as `EXTRA_GOFLAGS ?= -buildvcs=false` to
> 
> https://github.com/go-gitea/gitea/blob/389496bc93791b974246dc10aeccd2332873d6ce/Makefile#L68
> 
> ?
> It'll be overwritten if someone tries to add their own extra goflags, but I think that something that can be dealt with in a later PR.

It doesn't work!

**Comment by Gusted** [comment]
Why modifying the makefile? Using env variables is just easier:
```
EXTRA_GOFLAGS="-buildvcs=false" TAGS="sqlite sqlite_unlock_notify" make build
```

Which passes the go flag correctly:
```
CGO_CFLAGS="-g -O2 -DSQLITE_MAX_VARIABLE_NUMBER=32766" go build -v -buildvcs=false -tags 'sqlite sqlite_unlock_notify' -ldflags '-s -w  -X "main.MakeVersion=GNU Make 4.3" -X "main.Version=1.18.0+dev-58-g9d9bf66c3" -X "main.Tags=sqlite sqlite_unlock_notify"' -o gitea
```

**Comment by AbdulrhmnGhanem** [comment]
This doesn't resolve the issue addressed by this PR. Running the repro steps in the issue and adding `ENV EXTRA_GOFLAGS "-buildvcs=false"` to the docker file, it still fails. Also, in all previous trials above the flag gets passed correctly but it still fails.

**Comment by Gusted** [comment]
From my point of view, it's clear that we(the Gitea maintainers) don't see the benefit of adding this flag globally, as it doesn't provide any benefits to any other use-case. 

You can configure build flags like that via environments, there's no need to hard-code them. You mention that the flag is passed along with the build command, but it still fails, at that point it's out of Gitea's hand and likely a Go/Docker bug. The current PR doesn't fix the issue it still results into the same error. Unless this PR is to fix of why the error still exists even with `-buildvcs=false` passed, I don't see a reason to hold this PR open. 

Also as mentioned on the original issue, Go1.19 won't have this issue anymore.

**Pull request closed without merging**
